hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/tracers/AddOnEvmActionTracer.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.service.contract.impl.exec.tracers;
+
+import static java.util.Objects.requireNonNull;
+
+import com.hedera.hapi.streams.ContractActionType;
+import com.hedera.hapi.streams.ContractActions;
+import com.hedera.node.app.service.contract.impl.exec.ActionSidecarContentTracer;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.util.List;
+import java.util.Optional;
+import org.apache.tuweni.bytes.Bytes;
+import org.hyperledger.besu.datatypes.Transaction;
+import org.hyperledger.besu.evm.frame.ExceptionalHaltReason;
+import org.hyperledger.besu.evm.frame.MessageFrame;
+import org.hyperledger.besu.evm.log.Log;
+import org.hyperledger.besu.evm.operation.Operation;
+import org.hyperledger.besu.evm.tracing.OperationTracer;
+import org.hyperledger.besu.evm.worldstate.WorldView;
+
+/**
+ * A {@link OperationTracer} that delegates just the relevant callbacks to a {@link EvmActionTracer}, and all
+ * {@link OperationTracer} callbacks to a list of "add on" tracers.
+ */
+public class AddOnEvmActionTracer implements ActionSidecarContentTracer {
+    private final EvmActionTracer evmActionTracer;
+    private final List<OperationTracer> addOnTracers;
+
+    public AddOnEvmActionTracer(
+            @NonNull final EvmActionTracer evmActionTracer, @NonNull final List<OperationTracer> addOnTracers) {
```

Comment:
Looking around here confirms that we don't care about the ordering of the tracers.  Could, if interested, relax the collection type that's being passed around from `List`.  Probably not interested (or even interesting).

Reply:
Ah true, could be changed in future PR. 👍